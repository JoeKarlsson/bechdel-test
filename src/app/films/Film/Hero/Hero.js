import React from 'react';
import PropTypes from 'prop-types';
import './Hero.scss';

const Hero = props => {
	const { title, bechdelResults, images } = props;
	return (
		<div className="film-image">
			<img className="film-image" src={images.backdrop} alt={title} />
			<span className="film-hero-title">
				<h4>{title}</h4>
				<h3>Bechdel Pass: {bechdelResults.pass.toString().toUpperCase()}</h3>
				<p>Bechdel Score: {bechdelResults.bechdelScore} of 3</p>
			</span>
		</div>
	);
};

Hero.propTypes = {
	title: PropTypes.string.isRequired,
	bechdelResults: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
	images: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

Hero.defaultProps = {};

export default Hero;
