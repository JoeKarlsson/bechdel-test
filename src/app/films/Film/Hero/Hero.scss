.film-hero {
	height: 500px;
	overflow: hidden;
}

.film-image {
	width: 100%;
	position: relative;
}

.film-hero-title {
	position: absolute;
	top: 200px;
	left: 0;
	width: 100%;
	color: #fff;
	align-items: center;
	margin-left: 10%;
	background: rgba(0, 0, 0, 0.7);
	padding: 10px;
}
