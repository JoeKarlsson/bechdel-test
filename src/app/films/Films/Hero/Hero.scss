.section {
	padding: 8rem 0 7rem;
}

.section-heading,
.section-description {
	margin-bottom: 1.2rem;
}

.hero {
	background-image: url('../../../assets/images/hero.jpg');
	background-size: cover;
	color: #fff;
	display: flex;
	align-items: center;
	text-align: center;

	.hero-heading {
		text-align: center;
	}
}
