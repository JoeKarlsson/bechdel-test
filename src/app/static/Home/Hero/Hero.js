import React from 'react';
import { NavLink } from 'react-router-dom';
import './Hero.scss';

const Hero = () => (
	<div className="section hero">
		<div className="container">
			<div className="row">
				<div className="u-max-full-width ">
					<h4 className="hero-heading">bechdel.io</h4>
					<h5>find out if your favorite film passes the test.</h5>
					<NavLink to="/film/new">
						<button className="button">upload script</button>
					</NavLink>
				</div>
			</div>
		</div>
	</div>
);

export default Hero;
