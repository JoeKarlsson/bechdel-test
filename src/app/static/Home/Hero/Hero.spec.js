import React from 'react';
import Hero from './Hero';

describe('Hero Page', () => {
	describe('rendering', () => {
		it('match the snapshot', () => {
			const wrapper = shallow(<Hero />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
