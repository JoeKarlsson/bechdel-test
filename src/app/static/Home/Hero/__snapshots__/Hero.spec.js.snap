// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Page rendering match the snapshot 1`] = `ShallowWrapper {}`;
